packages/core/src/lib/connection_manager.ts

```diff
@@ -44,12 +52,57 @@ export class ConnectionManager {
     return instance;
   }
 
+  public async getPeersByDiscovery(): Promise<PeersByDiscovery> {
+    const peersDiscovered = await this.libp2p.peerStore.all();
+    const peersConnected = this.libp2p
```

Comment:
I guess we can optimize here and just `getConnections` which returns tags as well so that later there is no reason to await for peer store